operators/src/processing/raster_type_conversion.rs

```diff
@@ -0,0 +1,233 @@
+    async fn initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<dyn InitializedRasterOperator>> {
+        let input = self.sources.raster.initialize(context).await?;
+        let in_desc = input.result_descriptor();
+
+        let out_data_type = self.params.output_data_type;
+
+        let out_desc = RasterResultDescriptor {
+            spatial_reference: in_desc.spatial_reference,
+            data_type: out_data_type,
+            measurement: in_desc.measurement.clone(),
+            bbox: in_desc.bbox,
+            time: in_desc.time,
+        };
+
+        let initialized_operator = InitializedRasterTypeConversionOperator {
+            result_descriptor: out_desc,
+            source: input,
+        };
+
+        Ok(initialized_operator.boxed())
+    }
+}
+
+impl InitializedRasterOperator for InitializedRasterTypeConversionOperator {
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor> {
+        let source = self.source.query_processor()?;
+        let out_data_type = self.result_descriptor.data_type;
+
+        let res_op = call_on_generic_raster_processor!(source, source_proc => {
+            call_generic_raster_processor!(out_data_type,
+                RasterTypeConversionQueryProcessor::create_boxed(source_proc)
+            )
+        });
+
+        Ok(res_op)
+    }
+}
+
+pub struct RasterTypeConversionQueryProcessor<
+    Q: RasterQueryProcessor<RasterType = PIn>,
+    PIn: Pixel,
+    POut: Pixel,
+> {
+    query_processor: Q,
+    _p_out: std::marker::PhantomData<POut>,
+}
+
+impl<Q, PIn, POut> RasterTypeConversionQueryProcessor<Q, PIn, POut>
+where
+    Q: 'static + RasterQueryProcessor<RasterType = PIn>,
+    RasterTile2D<PIn>: ConvertDataType<RasterTile2D<POut>>,
+    PIn: Pixel,
+    POut: Pixel,
+{
+    pub fn new(query_processor: Q) -> Self {
+        Self {
+            query_processor,
+            _p_out: std::marker::PhantomData,
+        }
+    }
+
+    pub fn create_boxed(source: Q) -> Box<dyn RasterQueryProcessor<RasterType = POut>> {
+        RasterTypeConversionQueryProcessor::new(source).boxed()
+    }
+}
+
+#[async_trait]
+impl<Q, PIn: Pixel, POut: Pixel> QueryProcessor for RasterTypeConversionQueryProcessor<Q, PIn, POut>
+where
+    RasterTile2D<PIn>: ConvertDataType<RasterTile2D<POut>>,
+    Q: RasterQueryProcessor<RasterType = PIn>,
+{
+    type Output = RasterTile2D<POut>;
+    type SpatialBounds = SpatialPartition2D;
+
+    async fn query<'b>(
+        &'b self,
+        query: RasterQueryRectangle,
+        ctx: &'b dyn QueryContext,
+    ) -> Result<BoxStream<'b, Result<Self::Output>>> {
+        let stream = self.query_processor.raster_query(query, ctx).await?;
+        let converted_stream = stream.and_then(move |tile| {
+            crate::util::spawn_blocking(|| tile.convert_data_type()).map_err(Into::into)
```

Comment:
You implement this differently, previously it used `convert_data_type_parallel()`. Why?

Reply:
because we did no benchmarks back then. When i implemented the mask stuff i did a lot of benchmarks and they show that the overhead of a threadpool is to large for a simple operations like a cast, +, -, *, ...